alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -207,11 +207,25 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
 
         // Run through each backing index and apply appropriate mappings to query index
         indices.forEach { indexName ->
-            val concreteIndices = IndexUtils.resolveAllIndices(
+            var concreteIndices = IndexUtils.resolveAllIndices(
                 listOf(indexName),
                 monitorCtx.clusterService!!,
                 monitorCtx.indexNameExpressionResolver!!
             )
+            if (IndexUtils.isAlias(indexName, monitorCtx.clusterService!!.state()) ||
```

Comment:
if I use any alias, there can be multiple writeIndices. why are we assuming there can be only one writeIndex. That applies to datastream only

plz add tests for alias which points to multiple write indices and modify handling as required. 

I would explore modifying monitor metadata as required to handle aliases with multiple write indices.

Reply:
An `index/datastream alias` can have only `1 write index`. https://www.elastic.co/guide/en/elasticsearch/reference/current/aliases.html#write-index
`Common Problems` section - https://opster.com/guides/elasticsearch/glossary/elasticsearch-alias/
An `index pattern` can have multiple write indices though.